### PR TITLE
Modified timestamp in transfer status needs to be set for existing files too.

### PR DESCRIPTION
**Pull request by ylangisc**

### CODE REVIEW

File: googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveWriteFeature.java [hunk 1]
```diff
@@ -120,8 +121,13 @@ public File call(final AbstractHttpEntity entity) throws BackgroundException {
                         switch(postResponse.getStatusLine().getStatusCode()) {
                             case HttpStatus.SC_OK:
                                 if(status.isExists()) {
-                                    return session.getClient().getObjectParser().parseAndClose(
+                                    final File f = session.getClient().getObjectParser().parseAndClose(
                                             new InputStreamReader(postResponse.getEntity().getContent(), StandardCharsets.UTF_8), File.class);
+                                    if(null != status.getTimestamp()) {
+                                        new DriveTimestampFeature(session, fileid).setTimestamp(file, status.getTimestamp());
```
Comment (dkocher) [inline googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveWriteFeature.java:127]: I would duplicate the implementation which allows to directly return the updated `File` reference.

```suggestion
            final File properties = new File();
            properties.setModifiedTime(new DateTime(status.getTimestamp()));
            return session.getClient().files().update(fileid, properties).setFields(DriveAttributesFinderFeature.DEFAULT_FIELDS).
                    setSupportsAllDrives(new HostPreferences(session.getHost()).getBoolean("googledrive.teamdrive.enable")).execute();
```

**Review by dkocher — CHANGES_REQUESTED** [review]

File: googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveWriteFeature.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (dkocher) [reply googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveWriteFeature.java:127]: In general we do not handle requests gracefully but allow to retry the transfer.

Comment (AliveDevil) [reply googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveWriteFeature.java:127]: Create implementation method in Timestamp-feature that can be used from both Timestamp-feature and Write-feature, without duplicating code.

Comment (ylangisc) [reply googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveWriteFeature.java:127]: The `update` method must not contain fields that are not modifiable. The suggested change thus leads to 403 `The resource body includes fields which are not directly writable`.

**Review by AliveDevil — APPROVED** [review]

File: googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveWriteFeature.java (hunk 1 above)
Comment (dkocher) [inline googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveWriteFeature.java:127]:
```suggestion
                                        new DriveTimestampFeature(session, fileid).setTimestamp(file, status);
```

**Pull request merged**